### PR TITLE
0.6 compat

### PR DESCRIPTION
**Pull request by shashi**
tests run with no depwarns

### CODE REVIEW

**Comment by shashi** [comment]
Hold on, I might have a small optimization. Will update in a bit. Thanks!

**Comment by JeffBezanson** [comment]
F***ing Compat. I hate Compat so much.

File: src/columns.jl [hunk 1]
```diff
@@ -38,7 +40,7 @@ eltype{D,C}(::Type{Columns{D,C}}) = D
 length(c::Columns) = length(c.columns[1])
 ndims(c::Columns) = 1
 size(c::Columns) = (length(c),)
-linearindexing{T<:Columns}(::Type{T}) = Base.LinearFast()
+@compat Base.IndexStyle(::Type{<:Columns}) = IndexLinear()
```
Comment (tkelman) [inline src/columns.jl]: this was 0.19

Comment (shashi) [inline src/columns.jl]: Ok

**Pull request merged**